kura/examples/org.eclipse.kura.demo.modbus/src/main/java/org/eclipse/kura/demo/modbus/ModbusExample.java

```diff
@@ -149,68 +313,59 @@ private void configureDevice() throws ModbusProtocolException {
         }
     }
 
-    private void initializeLeds() throws ModbusProtocolException {
-        s_logger.debug("Initializing LEDs");	// once on startup, turn on each light
-        for (int led = 1; led <= 6; led++) {
-            this.m_protocolDevice.writeSingleCoil(this.m_slaveAddr, 2047 + led, true);
-            try {
-                Thread.sleep(200);
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
-            this.m_protocolDevice.writeSingleCoil(this.m_slaveAddr, 2047 + led, false);
+    private Properties getModbusProperties() {
+
+        if (this.m_properties == null) {
+            return null;
         }
-    }
 
-    private Properties getModbusProperties() {
         Properties prop = new Properties();
 
+        String modbusProtocol = (String) this.m_properties.get(MODBUS_PROTOCOL);
```

Comment:
An if condition will be helpful to check if the key is contained in the map or `this.m_properties.get(MODBUS_PROTOCOL) != null`.